contentful/client.ts

```diff
@@ -44,58 +44,53 @@ export interface Show {
   name?: string;
   start?: string;
   end?: string;
-  date?: string;
   description?: null | string;
-  picture?: Picture | null;
+  pictureUrl?: string | null;
   hosts?: null | string;
   producer?: null | string;
   color?: Color | null;
 }
-export interface Picture {
-  title?: string;
-  description?: null;
-  contentType?: string;
-  fileName?: string;
-  size?: number;
-  url?: string;
-  width?: number;
-  height?: number;
-}
 
-const fetchShowlist = async (
-  showlistId: string | string[]
-): Promise<{
-  showsByDate: Record<string, Show[]>;
-  weekKeys: Record<string, string[]>;
-}> => {
-  const data = await fetchContent<ShowlistQuery>(ShowlistDocument, {
-    showlistId,
-  });
+export const parseQueryResultToShowlist = (data: ShowlistQuery) => {
   const showsCollection =
     data.programmeCollection.items[0].showsCollection.items;
 
   const shows = showsCollection.map((item: any) => ({
     name: item.name,
     start: item.start,
     end: item.end,
-    date: format(new Date(item.start), 'y.M.dd'),
     description: item.description,
-    picture: item.picture,
+    pictureUrl: item.picture?.url || null,
     hosts: item.hosts,
     producer: item.producer,
     color: item.color,
   }));
+  return shows;
+};
 
+export const showsToGroups = (shows: Show[]) => {
   const showsByDate = groupBy(
     (day: any) => format(new Date(day.start), 'y.M.dd'),
     shows
   );
-
   const weekKeys = generateWeekObj(showsByDate);
-
   return { showsByDate, weekKeys };
 };
```

Comment:
Mehän ei taideta enää tarvita näitä Contentful ohjelmakartta -hommia kun kartat tulee nyt joko sheetsistä tai S3:sta, ja samassa formaatissa molemmista?

Reply:
Teen näistä vaikka oman pärrän, jossa siivotaan kaikki vanhat pois. Silloin tähän PRään jää tieto miten se on ennen tehty ja mitä graphqlliä tarvittu